### ISSUE TITLE
Does not work with newer openssl

### ISSUE THREAD

**Issue by krbvroc1** [dracut-crypt-ssh/dracut-crypt-ssh#3]
I cannot get this to work. I believe it is because the newer openssl stores RSA keys in PKCS8 format instead of the older PKCS1 format. So dropbearconvert fails and the dracut stuff is not installed.

I thought 'openssl rsa -in file.pem -out key.pem' would convert from new to old format but it doesn't using the current Centos 7.2 openssl. (openssl-1.0.1e-42.el7.9 currently)

**Comment by jpap** [comment]
The PR given here is a good one; would love to see it merged.

**Comment by rbu** [comment]
First of all, sorry for taking insanely long to respond to this.
@krbvroc1, thank you for the bug report and patch. Can you explain exactly what you tried to do and what did not work? In a fresh CentOS 7.3, I was able to get both "GENERATE" and "SYSTEM" values for dropbear_rsa_key / dropbear_ecdsa_key to work.

Were you trying to use the "/path/to/openssh_key" variant?

**Comment by krbvroc1** [comment]
@rbu - It has been many months since I reported this, let me revisit/refresh and get back to you. I have not yet upgraded to Centos 7.3 (came out just a few days ago) so maybe the problem was resolved in the openssl of Centos 7.3? From my fuzzy memory, yes, I was using the path variant and I tried to covert from an RSA key using the command listed in my initial report. The newer openssl did not output it in a format that dropbear could read.
